src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -381,6 +384,29 @@ BlockIO InterpreterDropQuery::executeToDatabase(const ASTDropQuery & query)
     return res;
 }
 
+bool matchesLikeWithRE2(const String & haystack,
```

Comment:
How about naming it something generic (without the lib name in the function)?
```suggestion
bool matchesLikePattern(const String & haystack,
```